.github/workflows/cd.yaml

```
@@ -6,240 +6,40 @@ on:
-          ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
-          ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
-
-          # Terraform variables.
-          TF_VAR_key_file: "../../key.pem"
-          TF_VAR_public_key_jwk_file: "../../key.public.jwk"
-          TF_VAR_acr_resource_group: ${{ secrets.COMMON_RESOURCE_GROUP }}
-          TF_VAR_acr_name: ${{ secrets.ACR_NAME }}
-          TF_VAR_participant_name: ${{ matrix.participant }}
-          TF_VAR_prefix: ${{ env.RESOURCES_PREFIX }}
-          TF_VAR_resource_group: rg-${{ matrix.participant }}-${{ github.run_number }}
-          TF_VAR_runtime_image: mvd-edc/connector:${{ github.run_number }}
-          TF_VAR_application_sp_object_id: ${{ secrets.APP_OBJECT_ID }}
-          TF_VAR_application_sp_client_id: ${{ secrets.APP_CLIENT_ID }}
-          TF_VAR_application_sp_client_secret: ${{ secrets.APP_CLIENT_SECRET }}
-          TF_VAR_registry_resource_group: ${{ secrets.COMMON_RESOURCE_GROUP }}
-          TF_VAR_registry_storage_account: ${{ secrets.REGISTRY_STORAGE_ACCOUNT }}
-          TF_VAR_registry_share: ${{ secrets.REGISTRY_SHARE }}
-
-      - uses: actions/upload-artifact@v3
-        name: 'Publish Backend conf'
-        with:
-          name: terraform-${{ matrix.participant }}-backend-conf
-          path: deployment/terraform/backend.conf
-          retention-days: 1
-
-      - name: 'Verify did endpoint is available'
-        run: curl $DID_ENDPOINT | jq '.id'
-
-      - name: 'Verify deployed EDC is healthy'
-        run: curl --retry 6 --fail http://${EDC_HOST}:8181/api/check/health
-
-      - name: 'Seed data'
-        run: |
-          npm install -g newman
-          deployment/seed-data.sh
-
-  Verify:
-    needs: Deploy
-    runs-on: ubuntu-latest
-    steps:
-      # Checkout MVD code
-      - uses: actions/checkout@v2
-
-      - uses: ./.github/actions/gradle-setup
-
-      - name: 'Az CLI login'
-        uses: azure/login@v1
-        with:
-          client-id: ${{ secrets.ARM_CLIENT_ID }}
-          tenant-id: ${{ secrets.ARM_TENANT_ID }}
-          subscription-id: ${{ secrets.ARM_SUBSCRIPTION_ID }}
-
-      - name: 'System tests'
-        run: |
-          ./gradlew :system-tests:test
-        env:
-          CONSUMER_MANAGEMENT_URL: http://${{ needs.Deploy.outputs.company1_edc_host }}:9191
-          PROVIDER_MANAGEMENT_URL: http://${{ needs.Deploy.outputs.company2_edc_host }}:9191
-          PROVIDER_IDS_URL: http://${{ needs.Deploy.outputs.company2_edc_host }}:8282
-          CONSUMER_KEY_VAULT: ${{ needs.Deploy.outputs.company1_key_vault }}
-          CONSUMER_CATALOG_URL: http://${{ needs.Deploy.outputs.company1_edc_host }}:8181/api/federatedcatalog
-          API_KEY: ${{ needs.Deploy.outputs.company1_api_key }}
+  Deploy:
+    uses: ./.github/workflows/deploy.yaml
+    secrets: inherit
+    needs: SetResourcesPrefix
+    with:
+      resources_prefix: ${{ needs.SetResourcesPrefix.outputs.resources_prefix }}
```

why not

```suggestion
      resources_prefix: ${{ env.CD_RESOURCES_PREFIX }}${{ github.run_number }}
```

and remove the job?

---

We can’t because of a bug, env.* does not work when you pass input to a called workflow from caller workflow
There is a comment explaining that on top of the SetResources job.

---

I see, I'd misunderstood the comment to mean that we can't use env in the called workflow itself. Made a proposal to clarify